### PR TITLE
Issue #8539: change diff trigger

### LINKED ISSUE #8539: Change diff report generation pipeline logic to trigger to single comment
At the moment there are 2 triggers
diff report - triggers generation of diff report for one or two configs
single report - triggers generation of report only for patch branch

Desired behavior
1. Trigger is changed to only one, with text "GitHub, generate report"
2. Report logic should be changed to
if description has 

- "New module config:" - than this config is taken to generate report only for patch branch
- "Diff Regression config:" - than config is taken to generate usual diff report
- "Diff Regression config:" and "Diff Regression patch config": - than first config is taken for master branch, second is for patch branch.

Also update of [documentation](https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#executing-generation) is necessary

### PR DESCRIPTION
**Pull request by strkkk**
Closes #8539

Example of usages (fake config and 1 project is used, only for demo)
1. New module report https://github.com/strkkk/checkstyle/pull/5
2. Diff report https://github.com/strkkk/checkstyle/pull/4
3. Diff report with different patch config https://github.com/strkkk/checkstyle/pull/6

PR for doc update 
https://github.com/checkstyle/contribution/pull/497

### CODE REVIEW

File: .github/workflows/diff_report.yml [hunk 1]
```diff
@@ -45,14 +45,20 @@ jobs:
...
+        sed 's/Diff Regression projects: //' temp > projects_link
+        echo ::set-output name=projects_link::$(cat projects_link)
+
         grep "^Diff Regression config:" text > temp || echo "" > temp
-        sed 's/Diff Regression config: //' temp > config
-        echo ::set-output name=config_link::$(cat config)
+        sed 's/Diff Regression config: //' temp > config_link
+        echo ::set-output name=config_link::$(cat config_link)
+
+        grep "^New module config:" text > temp || echo "" > temp
```
Comment (romani) [inline .github/workflows/diff_report.yml:55]: `grep "^New module config:" text | cat > temp` might be better form

File: .github/workflows/diff_report.yml [hunk 2]
```diff
@@ -64,17 +70,27 @@ jobs:
   make_report:
     runs-on: ubuntu-latest
     needs: parse_body
-    if: needs.parse_body.outputs.project_link != '' && needs.parse_body.outputs.config_link != ''
+    if: needs.parse_body.outputs.projects_link != '' &&
+        (needs.parse_body.outputs.config_link != '' ||
```
Comment (romani) [inline .github/workflows/diff_report.yml:74]: please move `&&` and `||` to the begining

File: .github/workflows/diff_report.yml [hunk 3]
```diff
@@ -114,29 +127,22 @@ jobs:
...
        run: |
          cd contribution/checkstyle-tester
          bash
          REF="origin/${{needs.parse_body.outputs.branch}}"
          REPO="../../checkstyle"
-         if [ -f patch_config.xml ]; then
-          groovy diff.groovy -r $REPO -b master -p $REF -bc config.xml -pc patch_config.xml\
+         if [ -f new_module_config.xml ]; then
+          groovy diff.groovy -r $REPO -p $REF -pc new_module_config.xml -m single\
+          -l project.properties
```
Comment (romani) [inline .github/workflows/diff_report.yml:138]: +2 spaces for indentation. The same for other linewraps.

File: .github/workflows/diff_report.yml [hunk 4]
```diff
@@ -45,14 +45,20 @@ jobs:
        id: parse
        run: |
         grep "^Diff Regression projects:" text > temp || echo "" > temp
-        sed 's/Diff Regression projects: //' temp > project
-        echo ::set-output name=project_link::$(cat project)
+        sed 's/Diff Regression projects: //' temp > projects_link
+        echo ::set-output name=projects_link::$(cat projects_link)
```
[the remark this answers is not in the document]
Comment (strkkk) [reply .github/workflows/diff_report.yml:49]: indentation correction and renaming is moved to separate commit

File: .github/workflows/diff_report.yml (hunk 1 above)
Comment (strkkk) [reply .github/workflows/diff_report.yml]: done

File: .github/workflows/diff_report.yml (hunk 2 above)
Comment (strkkk) [reply .github/workflows/diff_report.yml]: done

File: .github/workflows/diff_report.yml (hunk 3 above)
Comment (strkkk) [reply .github/workflows/diff_report.yml]: done in separate commit

**Comment by strkkk** [comment]
@romani I have made changes, links in first post contain examples of script work (updated version)
Doc PR is on its way, but should not be blocking for this PR

**Pull request merged**